### PR TITLE
Fix constr fetch resulting in invalid injection

### PR DESCRIPTION
**Pull request by bbrala**
Fixes: https://github.com/rectorphp/rector-symfony/issues/598

Proof of error: https://getrector.com/demo/b7387a86-a969-49c2-9ef5-3bf163f6e036

### CODE REVIEW

File: rules-tests/Symfony42/Rector/MethodCall/ContainerGetToConstructorInjectionRector/Fixture/constant_as_service_regression.php.inc [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+        /** @var \Drupal\g2\RouteFilter $rf */
+        $rf = $container->get(G2::SVC_ROUTE_FILTER);
+
+        /** @var \Drupal\g2\WOTD $wotd */
+        $wotd = $container->get(G2::SVC_WOTD);
+
+        return new static($entity_manager, $rf, $wotd, $configFactory);
+    }
+}
+?>
```
Comment (TomasVotruba) [inline rules-tests/Symfony42/Rector/MethodCall/ContainerGetToConstructorInjectionRector/Fixture/constant_as_service_regression.php.inc:213]: Thanks for the fix :+1: 

Could you make this test fixture as small as possible? So we can read it easily and see the buggy part :)

Comment (bbrala) [reply rules-tests/Symfony42/Rector/MethodCall/ContainerGetToConstructorInjectionRector/Fixture/constant_as_service_regression.php.inc:213]: Yeah can do later today. Should be easy enough

**Comment by bbrala** [comment]
I've updated the test to a minimal test that prooves the issue.

File: rules-tests/Symfony42/Rector/MethodCall/ContainerGetToConstructorInjectionRector/Fixture/constant_as_service_regression.php.inc (hunk 1 above)
Comment (bbrala) [reply rules-tests/Symfony42/Rector/MethodCall/ContainerGetToConstructorInjectionRector/Fixture/constant_as_service_regression.php.inc:25]: Minized the test

**Review by samsonasik — APPROVED** [review]

**Pull request merged**